builtin/credential/aws/path_role.go

```diff
@@ -348,14 +399,95 @@ func (b *backend) pathRoleCreateUpdate(
 		roleEntry.BoundIamInstanceProfileARN = boundIamInstanceProfileARNRaw.(string)
 	}
 
-	// Ensure that at least one bound is set on the role
-	switch {
-	case roleEntry.BoundAccountID != "":
-	case roleEntry.BoundAmiID != "":
-	case roleEntry.BoundIamInstanceProfileARN != "":
-	case roleEntry.BoundIamRoleARN != "":
-	default:
+	if boundIamPrincipalARNRaw, ok := data.GetOk("bound_iam_principal_arn"); ok {
+		roleEntry.BoundIamPrincipalARN = boundIamPrincipalARNRaw.(string)
+	}
+
+	if inferRoleTypeRaw, ok := data.GetOk("inferred_entity_type"); ok {
+		roleEntry.InferredEntityType = inferRoleTypeRaw.(string)
+	}
+
+	if inferredAWSRegionRaw, ok := data.GetOk("inferred_aws_region"); ok {
+		roleEntry.InferredAWSRegion = inferredAWSRegionRaw.(string)
+	}
+
+	// auth_type is a special case as it's immutable and can't be changed once a role is created
+	if authTypeRaw, ok := data.GetOk("auth_type"); ok {
+		if roleEntry.AuthType == "" {
+			switch authTypeRaw.(string) {
+			case ec2AuthType, iamAuthType:
+				roleEntry.AuthType = authTypeRaw.(string)
+			default:
+				return logical.ErrorResponse(fmt.Sprintf("unrecognized auth_type: %v", authTypeRaw.(string))), nil
+			}
+		} else if authTypeRaw.(string) != roleEntry.AuthType {
+			return logical.ErrorResponse("attempted to change auth_type on role"), nil
```

Comment:
[Optional] I'd phrase this as "changing `auth_type` on the role is not allowed".